### PR TITLE
[issue-15295] upgrade dependencies due to cves

### LINKED ISSUE #15295: update example dependencies due to cves
## Bug Report

https://github.com/apache/shardingsphere/blob/master/examples/pom.xml has a number of old dependencies

dependencies with cves:
* https://mvnrepository.com/artifact/mysql/mysql-connector-java
* https://github.com/advisories/GHSA-v7wg-cpwc-24m4 (postgresql)
* https://mvnrepository.com/artifact/junit/junit
* https://mvnrepository.com/artifact/org.mybatis/mybatis
* https://mvnrepository.com/artifact/com.h2database/h2
* https://mvnrepository.com/artifact/ch.qos.logback/logback-core

I did not update snakeyaml because it is being handled in a separate issue

### PR DESCRIPTION
**Pull request by pjfanning**
Fixes #15295

Changes proposed in this pull request:
- pom changes

### CODE REVIEW

File: examples/pom.xml [hunk 1]
```diff
@@ -45,14 +45,14 @@
         <spring-framework.version>5.2.19.RELEASE</spring-framework.version>
         <spring-boot.version>2.0.9.RELEASE</spring-boot.version>
         <hikari-cp.version>3.4.2</hikari-cp.version>
-        <mysql-connector-java.version>5.1.47</mysql-connector-java.version>
-        <postgresql.version>42.2.5</postgresql.version>
+        <mysql-connector-java.version>8.0.28</mysql-connector-java.version>
```
Comment (taojintianxia) [inline examples/pom.xml:48]: did you test the compatibility for these dependencies ?

File: examples/pom.xml [hunk 2]
```diff
@@ -63,7 +63,7 @@
         
         <seata.version>1.4.2</seata.version>
         
-        <junit.version>4.12</junit.version>
+        <junit.version>4.13.2</junit.version>
```
Comment (TeslaCN) [inline examples/pom.xml:66]: `assertThat` is marked as deprecated in junit 4.13. This require further consideration.

File: examples/pom.xml (hunk 1 above)
Comment (pjfanning) [reply examples/pom.xml:48]: @taojintianxia to be honest, I don't use shardingsphere - I am just an Apache member concerned about ASF projects relying on old dependencies with publicly disclosed attack vectors

Comment (pjfanning) [reply examples/pom.xml:48]: my experience is that all of these updates are low impact - that in most scenarios, they can be upgraded without breaking compatibility

File: examples/pom.xml (hunk 2 above)
Comment (linghengqian) [reply examples/pom.xml:66]: Maybe https://shardingsphere.apache.org/community/en/contribute/code-conduct/ should be updated?

Comment (pjfanning) [reply examples/pom.xml]: I reverted the junit piece - there is a CVE but it could dealt with in a separate issue

File: examples/pom.xml (hunk 1 above)
Comment (taojintianxia) [reply examples/pom.xml:48]: I see, anyway thanks for your contribution

File: examples/pom.xml (hunk 2 above)
Comment (linghengqian) [reply examples/pom.xml]: Refer to https://github.com/junit-team/junit4/pull/1150 , if the version is raised, it means a PR with keyword changes. (Similar to `org.junit.Assert.assertThat` is replaced by `org.hamcrest.MatcherAssert.assertThat` )

**Comment by linghengqian** [comment]
The logback and postgresql versions have been bumped up, no more changes needed.

**Comment by pjfanning** [comment]
@linghengqian I removed the mysql jar change because that seems to cause issues with tests - CI build passes now

**Review by linghengqian — APPROVED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**